File: JavaDemoGithubPractice/app/src/main/java/com/example/javademogithubpractice/mvp/contract/ILoginContract.java


package com.example.javademogithubpractice.mvp.contract;

import android.content.Intent;

import com.example.javademogithubpractice.mvp.model.BasicToken;


public interface ILoginContract {

    interface View extends IBaseContract.View{
        void onGetTokenSuccess(BasicToken basicToken);
        void onGetTokenError(String errorMsg);
        void onLoginComplete();
    }

    interface Presenter extends IBaseContract.Presenter<ILoginContract.View>{
        void getToken(String code, String state);
        String getOAuth2Url();
        void basicLogin(String userName, String password);
        void handleOauth(Intent intent);
        void getUserInfo(BasicToken basicToken);

    }

}

File: KotlinDemoGithubProject/app/src/main/java/com/example/kotlindemogithubproject/mvp/contract/ILoginContract.kt
package com.example.kotlindemogithubproject.mvp.contract

import android.content.Intent
import com.example.kotlindemogithubproject.mvp.model.BasicToken


interface ILoginContract {

    interface View : IBaseContract.View {

        fun onGetTokenSuccess(basicToken: BasicToken)

        fun onGetTokenError(errorMsg: String)

        fun onLoginComplete()

    }

    interface Presenter : IBaseContract.Presenter<View> {

        val oAuth2Url: String

        fun getToken(code: String, state: String)

        fun basicLogin(userName: String, password: String)

        fun handleOauth(intent: Intent)

        fun getUserInfo(basicToken: BasicToken)

    }

}
